flytekit/types/file/file.py

```diff
@@ -179,13 +179,21 @@ def extension(cls) -> str:
         return ""
 
     @classmethod
-    def new_remote_file(cls, name: typing.Optional[str] = None) -> FlyteFile:
+    def new_remote_file(cls, alt: typing.Optional[str] = None, name: typing.Optional[str] = None) -> FlyteFile:
         """
         Create a new FlyteFile object with a remote path.
+
+        :param alt: If you want to specify a different prefix head than the default one, you can specify it here.
+        :param name: If you want to specify a different name for the file, you can specify it here.
         """
         ctx = FlyteContextManager.current_context()
         r = name or ctx.file_access.get_random_string()
-        remote_path = ctx.file_access.join(ctx.file_access.raw_output_prefix, r)
+        pref = ctx.file_access.raw_output_prefix
+        if alt:
+            s_pref = pref.split("/")
+            s_pref[2] = alt
```

Comment:
let's make sure that `len(s_pref)` is at least 2 before executing this. I'm not sure if you intend to run this only in the case of blob stores (if that's the case we have to make this code more resilient and clarify its intent).

Reply:
Flawless only wants it for blob store, but I figured it could be used for any path that starts with `protocol://` where they'd want to change the first part. Behavior is slightly different in a local context, but still useful. This is mostly my open-question around how many fsspec protocols we should support..

Reply:
actually, scratch all that, the method [join in data persistence](https://github.com/flyteorg/flytekit/blob/d29e30eedc48f0f3b8d72dd5ea42f9dbdb1f2f13/flytekit/core/data_persistence.py#L435-L455) already does the correct thing. You should be able to replace all this code with this just:
```
remote_path = ctx.file_access.join(ctx.file_access.raw_output_prefix, alt, r)
```

Reply:
But it's part of the raw output prefix that we're trying to replace, specifically the bucket name. Unless I'm misunderstanding.